src/chainstate/stacks/db/blocks.rs

```diff
@@ -3392,8 +3396,8 @@ impl StacksChainState {
                     } else {
                         0
                     };
-                if total_spent > origin.stx_balance {
-                    return Err(MemPoolRejection::NotEnoughFunds(total_spent, origin.stx_balance))
+                if !origin.stx_balance.can_transfer(total_spent, 0) {
+                    return Err(MemPoolRejection::NotEnoughFunds(total_spent, origin.stx_balance.get_available_balance_at_block(0)))
```

Comment:
And here.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/0c6c1d9dd8e47e79643c1edea6dc95de404ed262